opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -778,9 +778,7 @@ namespace Opm
             well_state_copy.currentProductionControls()[index_of_well_] = Well::ProducerCMode::BHP;
         }
         well_state_copy.bhp()[well_copy.index_of_well_] = bhp;
-
-        well_copy.updatePrimaryVariables(well_state_copy, deferred_logger);
-        well_copy.initPrimaryVariablesEvaluation();
+        well_copy.calculateExplicitQuantities(ebosSimulator, well_state_copy, deferred_logger);
```

Comment:
It is understandable to calculate the explicit quantities here, which is based on the well_state_copy. Did you evaluate the benefit here or it solves some problems?

And also, why removing the update the primary variables? Because when well state changes, the primary variables need to be updated.

Reply:
Okay. I did not see `updatePrimaryVariables` was added to `calculateExplicitQuantities` some time ago. Although it should not be there. 

I take back my comment here. Sorry about that.